### PR TITLE
Add WARN Logging if Mock Network Accepts Huge Number of Connections

### PR DESCRIPTION
**Pull request by original-brownbear**
* As discussed, added warn logging to rule out endless accept loops for #43387
* Had to handle it by the relatively awkward override in the mock NIO
because we don't have logging in the NIO module where
(`ServerChannelContext` lives)

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

File: test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java [hunk 1]
```diff
@@ -228,12 +229,25 @@ public MockSocketChannel createChannel(NioSelector selector, SocketChannel chann
...
+            ServerChannelContext context = new ServerChannelContext(nioServerChannel, null, selector, null,
+                exceptionHandler) {
+                @Override
+                public void acceptChannels(Supplier<NioSelector> selectorSupplier) throws IOException {
+                    int acceptCount = 0;
+                    NioSocketChannel acceptedChannel;
+                    while ((acceptedChannel = MockTcpChannelFactory.this.acceptNioChannel(this, selectorSupplier)) != null) {
+                        acceptChannel(acceptedChannel);
+                        ++acceptCount;
+                        if (acceptCount > 0 && acceptCount % 100 == 0) {
```
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java]: Isn't acceptCount always > 0 given that you start with ' and incremented it in the line above?

**Review by ywelsch — APPROVED** [review]
Left one comment, looking good o.w.

File: test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java (hunk 1 above)
Comment (original-brownbear) [reply test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java]: Fair point, my bad :)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
thank @ywelsch + @tbrooks8

**Pull request merged**